src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java

```diff
@@ -1061,7 +1084,15 @@ public synchronized void handle(Message<T> event) {
         }
       } else {
         checkNextTick();
-        handler.handle(event);
+        MessageImpl abc = (MessageImpl) event;
+        metrics.beginHandleMessage(metric, abc.getSocket() == null);
+        try {
+          handler.handle(event);
+          metrics.endHandleMessage(metric, null);
```

Comment:
I see what you mean.

Your point is about detecting blocking code in the logs.

This provides the opportunity to the metrics implementation to report the precise behavior of the eventbus message delivery and log them in a time serie database to make queries.